components/usage/pkg/controller/reconciler.go

```diff
@@ -105,33 +105,69 @@ func (u *UsageReconciler) ReconcileTimeRange(ctx context.Context, from, to time.
 	}
 	log.WithField("workspace_instances", instances).Debug("Successfully loaded workspace instances.")
 
-	instancesByAttributionID := groupInstancesByAttributionID(instances)
+	usageRecords := instancesToUsageRecords(instances, u.pricer, now)
+	//instancesByAttributionID := groupInstancesByAttributionID(instances)
 
-	err = u.billingController.Reconcile(ctx, now, instancesByAttributionID)
+	err = u.billingController.Reconcile(ctx, usageRecords)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to reconcile billing: %w", err)
 	}
 
-	return status, instancesByAttributionID, nil
+	return status, usageRecords, nil
 }
 
-type UsageReport map[db.AttributionID][]db.WorkspaceInstanceForUsage
+func instancesToUsageRecords(instances []db.WorkspaceInstanceForUsage, pricer *WorkspacePricer, now time.Time) []db.WorkspaceInstanceUsage {
```

Comment:
These two types `db.WorkspaceInstanceForUsage` and `db.WorkspaceInstanceUsage` are very similarly named. Are there better names we can use?

Reply:
Suggestions welcome, I couldn't come up with one. The `ForUsage` is a join between WorkspaceInstance and Workspace